scripts/spread_calculation.py

```diff
@@ -0,0 +1,255 @@
+            
+            for trading_pair, price_data in bid_ask_prices.items():
+                bid = float(price_data['bid'])
+                ask = float(price_data['ask'])
+                mid_price = float(price_data['mid'])
+                spread = float(price_data['spread'])
+                spread_pct = float(price_data['spread_pct'])
+                
+                self.logger().info(
+                    f"{trading_pair} → BID: {bid}, ASK: {ask}, "
+                    f"SPREAD: {spread:.6f} ({spread_pct:.4f}%)"
+                )
+                
+                market_data_batch.append({
+                    'exchange': self.connector_name,
+                    'trading_pair': trading_pair,
+                    'best_bid': bid,
+                    'best_ask': ask,
+                    'mid_price': mid_price,
+                    'spread': spread,
+                    'spread_pct': spread_pct
+                })
+            
+            self.store_spread_data(market_data_batch)
+            self.logger().info(f"Processed {len(bid_ask_prices)} trading pairs from {self.connector_name}")
+            
+        except Exception as e:
+            self.logger().error(f"Error fetching bid/ask prices from {self.connector_name}: {e}")
+    
+    async def _fetch_using_get_prices(self):
+        """
+        Fallback method for rate sources that don't support get_bid_ask_prices.
+        Uses get_prices which only returns mid prices (no spread data).
+        """
+        try:
+            prices = await self._rate_source.get_prices(quote_token=self.quote_token)
+            
+            if not prices:
+                self.logger().warning(f"No prices received from {self.connector_name}")
+                return
+            
+            market_data_batch: List[dict] = []
+            
+            for trading_pair, mid_price in prices.items():
+                self.logger().info(f"{trading_pair} → MID: {float(mid_price)}")
+                
+                market_data_batch.append({
+                    'exchange': self.connector_name,
+                    'trading_pair': trading_pair,
+                    'best_bid': None,
+                    'best_ask': None,
+                    'mid_price': float(mid_price),
+                    'spread': None,
+                    'spread_pct': None
+                })
+            
+            self.store_spread_data(market_data_batch)
+            self.logger().info(f"Processed {len(prices)} trading pairs from {self.connector_name} (mid prices only)")
+            
+        except Exception as e:
+            self.logger().error(f"Error fetching prices from {self.connector_name}: {e}")
+    
+    def store_spread_data(self, market_data_list: List[dict]):
+        """
+        Store spread/market data using the MarketsRecorder.
+        """
+        if not market_data_list:
+            return
+            
+        try:
+            markets_recorder = MarketsRecorder._shared_instance
```

Comment:
```suggestion
            markets_recorder = MarketsRecorder.get_instance()
```
Also this part can be moved to __init__().